pkg/apis/core/validation/shoot.go

```diff
@@ -1154,6 +1155,23 @@ func ValidateClusterAutoscaler(autoScaler core.ClusterAutoscaler, fldPath *field
 	return allErrs
 }
 
+// ValidateCloudProfileReference validates the given CloudProfileReference fields.
+func ValidateCloudProfileReference(cloudProfileReference *core.CloudProfileReference, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if cloudProfileReference == nil {
+		return allErrs
+	}
+	if !sets.New("CloudProfile", "NamespacedCloudProfile").Has(cloudProfileReference.Kind) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("kind"), cloudProfileReference.Kind, "must be CloudProfile or NamespacedCloudProfile"))
+	}
```

Comment:
`NotSupported` seems to be more correct error to return here. You can do something like below
```suggestion
	cloudProfileReferenceKind := sets.New("CloudProfile", "NamespacedCloudProfile")
	if !cloudProfileReferenceKind.Has(string(cloudProfileReference.Kind)) {
		allErrs = append(allErrs, field.NotSupported(fldPath.Child("kind"), cloudProfileReference.Kind, sets.List(cloudProfileReferenceKind)))
	}
```